framework/src/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -1076,6 +1078,35 @@ public FlowExpressionContext copyAndUseOuterReceiver() {
         }
     }
 
+    /**
+     * Returns the innermost enclosing type of the given type, or Type.noType if no such type was
+     * found.
```

Comment:
Can you add some more explanation why this method is necessary?
Why is calling `DeclaredType.getEnclosingType` not the right thing in this case?
Why is this only an issue for static fields?
Why is this only an issue in `FlowExpressionParseUtil.java` and not in other uses of getEnclosingType?